src/agent/configuration_parser/src/configuration_parser.cpp

```diff
@@ -4,8 +4,31 @@
 
 namespace
 {
+#ifdef _WIN32
+    const std::string getConfigFilePath()
+    {
+        std::string configFilePath;
+        char* programData = nullptr;
+        std::size_t len = 0;
+        int error = _dupenv_s(&programData, &len, "ProgramData");
+
+        if (error || programData == nullptr)
+        {
+            configFilePath = "C:\\ProgramData\\wazuh-agent\\config\\wazuh-agent.yml";
+        }
+        else
+        {
+            configFilePath = std::string(programData) + "\\wazuh-agent\\config\\wazuh-agent.yml";
+        }
+        free(programData);
```

Comment:
Check that the variable is not nullptr.

Reply:
If the pointer is null free() does nothing.